marlowe-runtime/doc/ReadMe.md

```diff
@@ -1,100 +1,141 @@
+Each service handles rollbacks via the use of intersection points that reference specific slots/blocks on the blockchain. 
 
+Most of the data flow is stream-oriented and the services are as stateless as possible. The information flow within the backend maximizes the node as the single source of truth, thus minimizing the danger of downstream components receiving inconsistent information. 
 
-### Chain Seek Daemon
+The Haskell types in the client API for Runtime Clients are decoupled from dependencies upon the numerous Cardano packages for the ledger, node, plutus, etc., so that a Haskell client for Runtime has very few Runtime or Cardano dependencies in its `.cabal` file.
 
-The `chainseekd` executable provides services for querying the blockchain for information that may relate to Marlowe contracts. It manages and updates partitioned PostgreSQL tables for all blocks, transactions, inputs, outputs, and assets on the blockchain, not just for Marlowe transactions. It handles rollbacks in the blockchain. It syncs from the genesis block rapidly for test networks like `preprod` and `preview`, but currently takes about 15 hours to sync from the genesis of `mainnet`.
+See the [tutorial for the Marlowe Runtime command-line-interface](tutorial.md) or the [deployment instructions for the Marlowe Runtime backend](deployment.md) for more information.
 
-Three APIs, accessible over three different TCP sockets, are provided:
-- The *job* API provides one function:
-	- Submit a transaction to the local Cardano node.
-- The *query* API provides two functions:
-	- Query the local state of the Cardano node for information such as the system start time, the protocol parameters, or the slotting.
-	- Get the unspent UTxOs.
-- The *sync* API streams or replays the block and transaction information from the Cardano node.
+![Backend component services in Marlowe Runtime](diagrams/components.svg)
 
-See `chainseekd`'s [help page](chainseekd.md) or its [deployment instructions](../../marlowe-chain-sync/ReadMe.md) for more information.
 
+### Chain Seek Daemon (`chainseekd`)
 
-### Marlowe History
+- provides services for querying the blockchain for information that may relate to Marlowe contracts, 
+- manages and updates partitioned PostgreSQL tables for all blocks, transactions, inputs, outputs, and assets on the blockchain, not just for Marlowe transactions, 
+- syncs from the genesis block rapidly for test networks like `preprod` and `preview` ( N.B : currently takes about 15 hours to sync from the genesis of `mainnet`)
+- handles rollbacks in the blockchain
+- 3 APIs, each on different TCP sockets.
+	- *job* API: 
+		- submit a transaction to the local Cardano node
+	- *query* API:
+		- Query the local state of the Cardano node (e.g : system start time, the protocol parameters, or the slotting)
+		- Get the unspent UTxOs
+	- *sync* API: 
+		- streams or replays the blocks and transactions information from the Cardano node
```

Comment:
It's worth highlighting that this protocol (`ChainSeek`) does more than just replay blocks and transactions - it allows the client to skip directly to a block of interest in the future using a composable chain event query DSL.

Reply:
replaced it with : 

```
	- *sync* API: 
		- Streams or replays the blocks and transactions information from the Cardano node
		- Allows the client to skip directly to a block of interest in the future using a composable chain event query DSL.
```